api/src/main/java/org/openmrs/module/appointments/service/impl/AppointmentsServiceImpl.java

```diff
@@ -172,26 +172,37 @@ public Appointment getAppointmentByUuid(String uuid) {
     }
 
     @Override
-    public void changeStatus(Appointment appointment, String status, Date onDate) throws APIException{
-        if (!validateIfUserHasSelfOrAllAppointmentsAccess(appointment)) {
-            throw new APIAuthenticationException(Context.getMessageSourceService().getMessage("error.privilegesRequired",
-                    new Object[] { MANAGE_APPOINTMENTS }, null));
-        }
-        List<String> errors = new ArrayList<>();
+    public void changeStatus(Appointment appointment, String status, Date onDate) throws APIException {
         AppointmentStatus appointmentStatus = AppointmentStatus.valueOf(status);
+        throwExceptionIfNoProperPrivileges(appointment, appointmentStatus);
+        List<String> errors = new ArrayList<>();
         validateStatusChange(appointment, appointmentStatus, errors);
-        if(errors.isEmpty()) {
+        if (errors.isEmpty()) {
             appointment.setStatus(appointmentStatus);
             appointmentDao.save(appointment);
-            String notes = onDate != null ? onDate.toInstant().toString(): null;
+            String notes = onDate != null ? onDate.toInstant().toString() : null;
             createEventInAppointmentAudit(appointment, notes);
-        }
-        else {
+        } else {
             String message = StringUtils.join(errors, "\n");
             throw new APIException(message);
         }
     }
 
+    private void throwExceptionIfNoProperPrivileges(Appointment appointment, AppointmentStatus appointmentStatus) {
```

Comment:
Can we have a better name? The function should not be named "throw*". Should be named for intent - e.g. What are you validating - validateUserPrivilege() .. or something like that.

Reply:
Done